docs/mocksupport.rst

```diff
@@ -0,0 +1,131 @@
+.. _`Mock support`:
+
+Mock support
+============
+
+The pywbem PyPI package provides unit testing support for its users via its
+``pywbem_mock`` Python package. This package allows users of pywbem to
+define FakedWBEMConnections and CIM objects (classes, qualifier declarations,
+and instances) that will be processed by these connections. In effect this
+creates a fake WBEM Server with limited capabilities.
+
+The faked HMC environment is set up by creating an instance of the
+:class:`pywebm_mock.FakeWBEMConnection` class instead of the
+:class:`pywbem.WBEMConnection` class:
+
+The fake WBEMConnection is set up by creating an instance of the
+:class:`pywbem_mock.WBEMConnection` class instead of the
+:class:`pywbem_mock.WBEMConnection` class:
+
+.. code-block:: python
+
+    import pywbem
+    import pywbem_mock
+
+    conn = pywbem_mock.FakedWBEMConnection()
+
+    ... Add server CIM object resources (in this case some classes)
+
+    classes = conn.EnumerateClasses();
+    . . .
+
+Generally this mock environment supports:
+
+1. All of the WBEMConnection request methods that communicate with the WBEM
+server.
+
+2. Multiple namespaces.
+
+3. TODO
+
+It does this by mocking WBEM Server responses at the common level below the
+WBEMConnection methods such as GetClass, etc. but before the requests are
+formatted into HTTP.  This allows a mock repository to built by the user
+into which the data is inserted for any gived test.
+
+It does NOT support:
+
+1. CIM/XML protocol security
+2. Dynamic WBEM Server providers
+3. It does not support processing of the queries defined for ExecQuery in
+languages like CQL and WQL.  The mocker just returns found instances.
+4. It does not support the filter processing of the FQL filter query language
+   used by the Open... operations.  It returns the same data as if the filter
+   did not exist.
+5. It does not support time delay for operations.  The responses are returned
```

Comment:
This is written as if "time delay" was a feature of the normal connection. Maybe we should turn this into a note pointing out that the timing is faster.

Does the fake conn. support time statiatics, debugging via the last_request etc., logging? Any lack of such support would also be worth mentioning.

Reply:
Just removed 5. Note that I added a delay mechanism (delays) for defined number of milliseconds.

Noted that it does support statistics gathering but does not supply date to the last_request  or http logging.